docs/proposal-platforms.md

```diff
@@ -235,31 +235,183 @@ suite-run functionality:
+       shares the same filesystem as the scheduler host - we can remove this
+       functionality. Instead we should check for the existence of a client key
+       and fail the remote-init if one is found for a different `install target`
+       (different targets used within the same workflow should not share the
+       same run directory).
+   * Currently, the files which need to be installed are transferred as a tar
+     file via stdin. This has the advantage that the remote init can be done as
+     a single ssh. We will change the installation to use rsync which will
+     require a second ssh but this will be better for installing more files and
+     allow the logging to report which files have been changed after a reload.
+     * If we change cylc to re-use ssh connections by default (using
+       `ControlPersist`) then this will avoid the ssh overhead (and will
+       benefit other ssh connections as well).
+     * We will probably need a third ssh connection for transferring the client
+       public key.
+   * The remote installation will include particular files/directories.
+     * Items to include:
+       `.service/server.key .service/contact app/ bin/ etc/ lib/`.
+     * We will use the `--delete` rsync option so that any files removed from
+       from installed directories also get removed from the install targets on
+       reload/restart.
+     * Does this need to be configurable? (i.e. configure additional items to
+       include)? If so, global and/or workflow level?
+       * Is there any reason why workflow designers can't just stick to the list
+         of supported directories for files that need to be installed?
+       * Are there many existing Rose suites which rely on other directories
+         being installed? A quick search implies there are quite a few suites
+         using a wide range of sub-directories such as `python`, `util`, `data`
+         (although it is not clear how many of these would need to be installed
+         on remote platforms).
+       * Proposal: we allow the include list to be configurable at the global
+         level (to cater for old suites which need this) but encourage users to
+         adapt their workflows to use the supported directories and, therefore,
+         don't allow this to be configured at the suite level.
+     * The alternative is to use exclude rather than include and make sure we
+       exclude any file / directories which are generated at run time or target
+       specific (`log/ share/ work/` + various files / directories in
+       `.service/`).
+       * There are some suites around which write to top level directories (not
+         sure if this is deliberate) - exclude would be very bad in these cases.
+         Therefore, include is the safer option.
+
+2. Support moving some directories to different locations with symlinks to the
+   original location.
+   * New platform settings:
```

Comment:
I've gone for `[symlink dir]` - alternative suggestions welcome.

Reply:
Would `symlink path` be more accurate?